text/trace/0235-sampling-threshold-in-trace-state.md

```diff
@@ -0,0 +1,99 @@
+# Sampling Threshold Propagation in TraceState
+
+## Motivation
+
+Sampling can theoretically take place at nearly any point in a distributed tracing system. If sampling is to be performed at multiple points in the process, the only way to reason about it effectively is to make sure that the sampling decisions are **consistent**.
+In this context, consistency means that a positive sampling decision made at a particular point with probability p1 implies a positive sampling decision made at another point that samples a different piece of information from the same trace with probability p2 >= p1.
+
+## Explanation
+
+The existing, experimental [specification for probability sampling using TraceState](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/tracestate-probability-sampling.md) is limited to powers-of-two probabilities, and is designed to work without making assumptions about TraceID randomness.
+This system can only achieve non-power-of-two sampling using interpolation between powers of two, which is unnecessarily restrictive.
+In existing sampling systems, sampling probabilities like 1%, 10%, and 75% are common, and it should be possible to express these without interpolation.
+There is also a need for consistent sampling in the collection path (outside of the head-sampling paths) and using inherent randomness in the traceID is a less-expensive solution than referencing a custom `r-value` from the tracestate in every span.
+This proposal introduces a new value with the key `th` as a replacement for the `p` value in the previous specification.
+The `p` value is limited to powers of two, while this proposal is not.
+This allows for the continued expression of randomness using `r-value` as specified there using the key `r`.
+To distinguish the cases, this proposal uses the key `rv`.
+
+In order to make consistent sampling decisions across the entire path of the trace, two values SHOULD be propagated with the trace:
+
+1. A _random_ (or pseudo-random) 56-bit value, called `R` below.
+2. A 56-bit trace _threshold_ as expressed in the TraceState, called `T` below.
+
+The sampling decision is propagated with the following algorithm:
+
+* If the `th` key is not specified, Always Sample.
+* Else derive `T` by parsing the `th` key as a hex value as described below.
+* If `T` is 0 and the _sampled_ flag is set, Always Sample. This implies that non-probabilistic sampling is taking place.
+* Compare the 56 bits of `T` with the 56 bits of `R`. If `T <= R`, then do not sample.
```

Comment:
Isn't 'T' encoding the sampling probability of the parent?  Since we are talking about consistent probability sampling here, the callee could have a different sampling probability, correct? If so, shouldn't it be comparing **_its_** sampling probability (expressed in a similar 56 bits) with the 56 bits of 'R' value to make the consistent sampling decision?  This is based on my understanding that 'th' is the replacement for p-value (and hence encodes the adjusted count of the parent, but doesn't reflect the callee's sampling probability).

Reply:
Yes, this only describes how the parent's threshold is determined. What is missing is that each sampler is completely free to choose its own sampling probability, but can make its choice based on the parent's threshold. In this way, it is possible, for example, to implement parent-based sampling by setting the threshold equal to that of the parent.

Reply:
There is a parallel discussion https://github.com/open-telemetry/opentelemetry-proto/issues/507 and https://github.com/open-telemetry/opentelemetry-proto/pull/484. 

Summarizing, I believe the Span's trace-state incorporates changes made by the span, considering the parent's input. We are discussing in those issues whether to add a bit to say whether the parent was sampled or not, since the span's flags refer to itself. However, I would not propose to record the parent's tracestate with the span, in addition to the span's tracestate.

In these abstract formulas, it doesn't matter which is which. There will be samplers that use the parent's probability to set the child's probability, among other approaches. The specification will say how the span should modify the parent's trace state to set it's own, as was done in https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/tracestate-probability-sampling.md.